utils.py

```diff
@@ -102,4 +102,40 @@ def get_model_coefficients(classifier, feature_set, covariate_names):
     coef_df['feature_set'] = feature_set
     
     
-    return coef_df
\ No newline at end of file
+    return coef_df
+
+def get_genes_coefficients(pca_object, classifier_object,
+                           expression_df):
+    """Identify gene coefficients from classifier after pca.
+
+    Args:
+        pca_object: The pca object from running pca on the expression_df.
+        classifier_object: The logistic regression classifier object.
+        expression_df: The original (pre-pca) expression data frame.
+
+    Returns:
+        gene_coefficients: A dataframe with entreze gene-ID, coefficient and
+                            abbsolute value of coefficient. The dataframe is
+                            sorted by absolute value of coefficient.
+    """
+
+    import pandas as pd
+
+    # Get the classifier coefficients.
+    coefficients = classifier_object.coef_
+    coefficients = pd.DataFrame(coefficients).T
+    # Get the pca weights.
+    weights = pca_object.components_
+    headers = list(expression_df)
+    weights = pd.DataFrame(weights, columns=headers)
+    # Combine the coefficients and weights.
+    combined_df = pd.DataFrame(weights.values*coefficients.values,
```

Comment:
@patrick-miller can you explain more how matrix multiplication would be used here?

Also note that python has a matrix multiplication operator since 3.5 (`@`), which perhaps you could use.